client.go

```diff
@@ -768,22 +783,37 @@ func (c *TunaSessionClient) handleMsg(conn *Conn, sess *ncp.Session, i int) erro
 	return nil
 }
 
-func (c *TunaSessionClient) handleConn(conn *Conn, sessKey string, i int) {
+func (c *TunaSessionClient) handleConn(conn *Conn, remoteAddr string, sessionID []byte, i int) {
 	if conn == nil {
 		return
 	}
+
+	sessKey := sessionKey(remoteAddr, sessionID)
 	c.Lock()
 	sess := c.sessions[sessKey]
 	if sess == nil {
 		c.Unlock()
 		return
 	}
 	c.connCount[sessKey]++
+	tcpCount, ok := c.addrConnCount[remoteAddr]
+	if ok {
+		tcpCount[conn.RemoteAddr().String()]++
+	} else {
+		c.addrConnCount[remoteAddr] = make(map[string]int)
+		c.addrConnCount[remoteAddr][conn.RemoteAddr().String()]++
+	}
 	c.Unlock()
 
 	defer func() {
 		c.Lock()
 		c.connCount[sessKey]--
+		c.addrConnCount[remoteAddr][conn.RemoteAddr().String()]--
+		if c.addrConnCount[remoteAddr][conn.RemoteAddr().String()] <= 0 { // When any count drops to 0 we remove c.pubAddrs[remoteAddr] from cache
+			delete(c.pubAddrs, remoteAddr)
```

Comment:
Can `delete(c.addrConnCount[remoteAddr], conn.RemoteAddr().String())` too. 
Also we can `delete(c.addrConnCount, remoteAddr)` if `len(c.addrConnCount[remoteAddr]) == 0`